sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2602,7 +2602,7 @@ class Dataset[T] private[sql](
    * @since 2.0.0
    */
   def inputFiles: Array[String] = {
-    val files: Seq[String] = logicalPlan.collect {
+    val files: Seq[String] = queryExecution.optimizedPlan.collect {
```

Comment:
We only determine the partitions read after optimization, so it's necessary to read it from that instead of the logical plan.